contracts/button/lib.rs

```diff
@@ -295,6 +298,80 @@ mod button_game {
             contract
         }
 
+        fn reset_state(&mut self) {
+            let now = self.env().block_number();
+
+            self.presses = 0;
+            self.last_presser = None;
+            self.last_press = now;
+            self.total_rewards = 0;
+
+            Self::emit_event(self.env(), Event::GameReset(GameReset { when: now }));
+        }
+
+        fn reward_pressiah(&self) -> ButtonResult<()> {
+            if let Some(pressiah) = self.last_presser {
+                let reward = self.pressiah_score();
+                self.mint_reward(pressiah, reward)??;
+            };
+
+            Ok(())
+        }
+
+        fn ensure_dead(&self) -> ButtonResult<()> {
+            if !self.is_dead() {
+                return Err(GameError::BeforeDeadline);
+            } else {
+                Ok(())
+            }
+        }
+
+        fn dump_tickets_in_marketplace(&self) -> ButtonResult<()> {
```

Comment:
putting these two into one function seems like an overkill, and it is "semantically incorrect" : it transfers (aka dumps) tickets but it also re-sets the contract state

Reply:
I disagree, but not strongly enough to argue :P